src/CoreShop/Bundle/StorageListBundle/DependencyInjection/CoreShopStorageListExtension.php

```diff
@@ -109,10 +109,13 @@ public function load(array $configs, ContainerBuilder $container): void
                 $container->setDefinition('coreshop.storage_list.session_subscriber.' . $name, $sessionSubscriber);
             }
 
-            $cacheSubscriber = new Definition(CacheListener::class, [
-                new Reference(PimcoreContextResolver::class),
-                new Reference($contextCompositeServiceName),
-            ]);
+            if ($list['disable_caching']) {
+                $cacheSubscriber = new Definition(CacheListener::class, [
+                    new Reference($list['resource']['repository']),
+                    new Reference($list['resource']['item_repository'])
+                ]);
+            }
+            
             $cacheSubscriber->addTag('kernel.event_subscriber');
 
             $container->setDefinition('coreshop.storage_list.cache_subscriber.' . $name, $cacheSubscriber);
```

Comment:
The next two statements should also be inside the condition, right?

```suggestion
            if ($list['disable_caching']) {
                $cacheSubscriber = new Definition(CacheListener::class, [
                    new Reference($list['resource']['repository']),
                    new Reference($list['resource']['item_repository'])
                ]);
            
                $cacheSubscriber->addTag('kernel.event_subscriber');

                $container->setDefinition('coreshop.storage_list.cache_subscriber.' . $name, $cacheSubscriber);
            }
```

Reply:
@jdreesen yes, I shouldn't commit changes after 8pm anymore 🙈